### PR TITLE
ASoC:SOF:skl enable the ioc

### PR DESCRIPTION
**Pull request by zhuyingjiang**
enabled the ioc, and fix BAR bug, the firmware status now can get

Signed-off-by: Zhu Yingjiang <yingjiang.zhu@linux.intel.com>

### CODE REVIEW

File: sound/soc/sof/intel/hda-dsp.c [hunk 1]
```diff
@@ -209,6 +209,19 @@ bool hda_dsp_core_is_enabled(struct snd_sof_dev *sdev,
...
+	/* stall core */
+	snd_sof_dsp_update_bits_unlocked(sdev, HDA_DSP_BAR,
+					 HDA_DSP_REG_ADSPCS,
+					 HDA_DSP_ADSPCS_CSTALL_MASK(core_mask),
+					 HDA_DSP_ADSPCS_CSTALL_MASK(core_mask));
+
+	/* set reset state */
+	return hda_dsp_core_reset_enter(sdev, core_mask);
+}
+
```
Comment (keyonjie) [inline sound/soc/sof/intel/hda-dsp.c:224]: we may need declaration in hda.h also?

File: sound/soc/sof/intel/hda-loader.c [hunk 2]
```diff
@@ -414,12 +413,12 @@ static int cl_dsp_init_skl(struct snd_sof_dev *sdev)
...
 
 	/* enable IPC DONE interrupt */
-	snd_sof_dsp_update_bits(sdev, HDA_DSP_BAR, chip->ipc_ctl,
+	snd_sof_dsp_update_bits(sdev, HDA_DSP_BAR, HDA_DSP_REG_HIPCCTL,
 				HDA_DSP_REG_HIPCCTL_DONE,
 				HDA_DSP_REG_HIPCCTL_DONE);
 
 	/* enable IPC BUSY interrupt */
-	snd_sof_dsp_update_bits(sdev, HDA_DSP_BAR, chip->ipc_ctl,
+	snd_sof_dsp_update_bits(sdev, HDA_DSP_BAR, HDA_DSP_REG_HIPCCTL,
```
Comment (keyonjie) [inline sound/soc/sof/intel/hda-loader.c]: I think we can still use chip->ipc_ctl here to make the code more comment with cl_dsp_init()?

Comment (zhuyingjiang) [reply sound/soc/sof/intel/hda-loader.c]: It is OK to use chip->ipc_ctl, but the previous enable core process is not the same

File: sound/soc/sof/intel/hda-dsp.c (hunk 1 above)
Comment (zhuyingjiang) [reply sound/soc/sof/intel/hda-dsp.c:224]: Yes, I will add that

**Comment by zhuyingjiang** [comment]
@keyonjie @lgirdwood

File: sound/soc/sof/intel/hda-loader.c [hunk 3]
```diff
@@ -375,7 +375,6 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 
 static int cl_dsp_init_skl(struct snd_sof_dev *sdev)
 {
-	const struct sof_intel_dsp_desc *chip = sdev->hda->desc;
```
Comment (keyonjie) [inline sound/soc/sof/intel/hda-loader.c]: I believe you haven't tested(or even compile) your update? removing 'chip' here doesn't compile?

**Comment by lgirdwood** [comment]
@zhuyingjiang pls ping when you pushed version that compiles.

**Comment by zhuyingjiang** [comment]
@lgirdwood

**Pull request merged**